### PR TITLE
Improving KExpressions

### PR DESCRIPTION
**Pull request by momosetkn**
Remove redundant nullables in KExpressions.

`concat` Expression arguments were required not-null.
https://github.com/domaframework/doma/blob/c8833a93cef4dbde6a81bc35426a3a5b5a8214c6/doma-core/src/main/java/org/seasar/doma/jdbc/criteria/expression/Expressions.java#L202-L205

`min` Expression arguments were required not-null.
https://github.com/domaframework/doma/blob/c8833a93cef4dbde6a81bc35426a3a5b5a8214c6/doma-core/src/main/java/org/seasar/doma/jdbc/criteria/expression/Expressions.java#L270-L272

`sum` Expression arguments were required not-null.
https://github.com/domaframework/doma/blob/c8833a93cef4dbde6a81bc35426a3a5b5a8214c6/doma-core/src/main/java/org/seasar/doma/jdbc/criteria/expression/Expressions.java#L276-L278

### CODE REVIEW

File: doma-kotlin/src/main/kotlin/org/seasar/doma/kotlin/jdbc/criteria/expression/KExpressions.kt [hunk 1]
```diff
@@ -200,13 +200,6 @@ object KExpressions {
         return Expressions.concat(left, right)
     }
 
-    fun <PROPERTY> concat(
```
Comment (nakamura-to) [inline doma-kotlin/src/main/kotlin/org/seasar/doma/kotlin/jdbc/criteria/expression/KExpressions.kt:203]: Could you please explain why this method needs to be removed?

Comment (momosetkn) [reply doma-kotlin/src/main/kotlin/org/seasar/doma/kotlin/jdbc/criteria/expression/KExpressions.kt:203]: Sorry, I deleted it by mistake. I revert this removing.
And I fixed test-code to meet `concat` method test coverage.

**Comment by nakamura-to** [comment]
@momosetkn  Could you make the same fixes that you made in commit https://github.com/domaframework/doma/pull/1142/commits/528fac67e8850de3ce2a5c49269fd7baf4d07d62 ?

**Comment by momosetkn** [comment]
@nakamura-to sorry, I fixed!

**Pull request merged**